lib/util/newDefaultObject.js

```diff
@@ -0,0 +1,31 @@
+import isEmpty from './isEmpty';
+
+function isArrayIndex(n) {
+  return /^[0-9]+$/.test(n);
+}
+
+function isArrayUpdate(updates) {
+  for (const updateKey of Object.keys(updates)) {
+    if (!isArrayIndex(updateKey)) { return false; }
+  }
+
+  return true;
+}
+
+function arrayOrObject(updates) {
+  if (!isEmpty(updates) && isArrayUpdate(updates)) {
+    return [];
+  }
+
+  return {};
+}
+
+function newDefaultObject(object, updates) {
+  if (typeof object === 'undefined' || object === null) {
+    return arrayOrObject(updates);
+  }
+
+  return object;
+}
+
+export default newDefaultObject;
```

Comment:
I might just call this `defaultObject`... it doesn't always new something up, but it does always "default" it.